frontend/packages/knative-plugin/src/components/test-function/TestFunctionModal.tsx

```diff
@@ -27,59 +21,92 @@ type Props = FormikProps<FormikValues> & TestFunctionModalProps & ModalComponent
 
 const TestFunctionModal: React.FC<Props> = (props) => {
   const { t } = useTranslation();
-  const { handleSubmit, cancel, close, isSubmitting, status, service, setFieldValue } = props;
+  const { handleSubmit, cancel, close, isSubmitting, service, setFieldValue } = props;
+  const [currentView, setCurrentView] = React.useState(ModalPanel.Request);
   const { name, namespace } = service.data.metadata;
   const [svc, loaded] = useK8sGet<ServiceKind>(ServiceModel, name, namespace);
 
   React.useEffect(() => {
     if (loaded) {
       setFieldValue('endpoint.url', svc.status.url);
+      svc.status.url?.endsWith('svc.cluster.local')
+        ? setFieldValue('endpoint.private', true)
+        : setFieldValue('endpoint.private', false);
     }
   }, [loaded, svc, setFieldValue]);
```

Comment:
This url and private props aren't used anymore. Can we remove it?

Reply:
Done :)